hash/keccakf1600x4/example_test.go

```diff
@@ -0,0 +1,80 @@
+package keccakf1600x4_test
+
+import (
+	"github.com/cloudflare/circl/hash/keccakf1600x4"
+
+	"encoding/binary"
+	"fmt"
+
+	// Vendored copy of golang.org/x/crypto/sha3.
+	sha3 "github.com/cloudflare/circl/internal/shake"
+)
+
+func Example() {
+	// As an example, computes the (first 32 bytes of a) SHAKE-256 stream of
+	// four short strings at the same time.
+	// See sign/dilithium/mode3/internal/sample.go for a real world example.
+
+	msgs := [][]byte{
+		[]byte("These are some short"),
+		[]byte("strings of the same "),
+		[]byte("length that fit in a"),
+		[]byte("single block.       "),
+	}
+	var hashes [4][32]byte
+
+	if !keccakf1600x4.Available {
+		// Compute hashes separately using golang.org/x/crypto/sha3 instead
+		// when a fast four-way implementation is not available.  A generic
+		// keccakf1600x4 implementation is quite a bit slower than using
+		// the non-interleaved hashes because of the need to interleave and
+		// deinterleave the state.
+		for i := 0; i < 4; i++ {
+			h := sha3.NewShake256()
+			_, _ = h.Write(msgs[i])
+			_, _ = h.Read(hashes[i][:])
+		}
+	} else {
+		// f1600 acts on 1600 bits arranged as 25 uint64s.  Our fourway f1600
+		// acts on four interleaved states; that is a [100]uint64.  (A separate
+		// type is used to ensure that the encapsulated [100]uint64 is aligned
+		// properly to be used efficiently with vector instructions.)
+		var perm keccakf1600x4.State
+		state := perm.Initialize()
+
+		// state is initialized with zeroes.  As the messages fit within one
+		// block, we only need to write the messages, domain separators
+		// and padding.
+		for i := 0; i < 4; i++ {
```

Comment:
This logic can be abstrated by a function beloging to the package. 
It could be a function that receives four pointers and loads them into the internal state.

Reply:
But that will be a trap!  I sped up SPHINCS+ by a factor of 2 precisely by removing such a function and doing it ad hoc.

Reply:
And that is in C.  In Go the difference will be more stark.

https://github.com/sphincs/sphincsplus/pull/9